CHANGELOG.md

```diff
@@ -4,6 +4,12 @@
 
 ### Snowpark Python API Updates
 
+#### Bug Fixes
+
+- Fixed a bug for the following functions that raised errors `.cast()` is applied to their output
+  - `from_json`
+  - `size`
```

Comment:
I still don't understand why the internal alias would cause failure to cast.. is it related to the incorrect column name?

Reply:
When we apply any other function on top of an aliased column, for example cast, the sql generated is malformed. For example `session.sql("select 1 as A").select(col("A").alias("A_alias").cast(StringType()))` generates
```
SELECT  CAST ("A" AS "A_ALIAS" AS STRING) FROM (select 1 as A)
```

This raises error. Therefore for functions implemented internally like listed here which add an `_alias`, if any operation is applied to the resulting column, we must ignore the alias and apply the function. More details are in doc attached in the JIRA

Reply:
oh I see, this example is clear enough to understand.